cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/BaseOAuthWrapperController.java

```diff
@@ -106,24 +107,28 @@ protected OAuthWebApplicationService createService(final RegisteredService regis
      * Create an authentication from a user profile.
      *
      * @param profile the given user profile
+     * @param service the registered service
      * @return the built authentication
      */
-    protected Authentication createAuthentication(final UserProfile profile) {
```

Comment:
I see your point, but I don't think it's feasible. The `createAuthentication method is called twice: in the `OAuth20AuthorizeController` controller and in the `OAuth20AccessTokenController` controller.

In the `OAuth20AuthorizeController` controller, the service ticket has already been consumed by the pac4j client to finish the authentication process for the callback service. There is no ST to validate here.
In the `OAuth20AccessTokenController` controller, we either retrieve the token and authentication or get the pac4j user profile and build the authentication. In this latter case, we don't have any ST, nor token.